### ISSUE TITLE
The HttpClient send request via HTTP/2 throw a exception

### ISSUE THREAD

**Issue by Jamesxql** [dotnet/corefx#16379]
On Ubuntu 14.04

my curl version is:
``` c#
curl 7.53.0 (x86_64-pc-linux-gnu) libcurl/7.53.0 OpenSSL/1.0.1f zlib/1.2.8 nghttp2/1.20.0-DEV librtmp/2.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp smb smbs smtp smtps telnet tftp 
Features: IPv6 Largefile NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy
```

And I send a request to HTTP/2 web, I receive this:
``` c#
ubuntu@ip-172-168-10-232:~$ curl --http2 -I https://http2.akamai.com/
HTTP/2 200 
server: Apache
etag: "5e9139516a4f3d7a27577441623ca90d:1465962509"
last-modified: Wed, 15 Jun 2016 03:48:29 GMT
accept-ranges: bytes
content-length: 11384
push: true
rtt: 159
ghost_ip: 23.201.177.105
ghost_service_ip: 2.18.214.61
client_real_ip: 52.80.17.185
client_ip: 52.80.17.185
myproto: h2
protocol_negotiation: h2
cache-control: max-age=43200
expires: Wed, 22 Feb 2017 23:45:37 GMT
date: Wed, 22 Feb 2017 11:45:37 GMT
content-type: textml;charset=UTF-8
accept-ch: DPR, Width, Viewport-Width, Downlink, Save-Data
access-control-max-age: 86400
access-control-allow-credentials: false
access-control-allow-headers: *
access-control-allow-methods: GET,HEAD,POST
access-control-allow-origin: *
strict-transport-security: max-age=31536000 ; includeSubDomains
```

But when I use HttpClient send this request, the program throw a exception:
```c#
Unhandled Exception: System.Net.Http.HttpRequestException: The server returned an invalid or unrecognized response.
   at System.Net.Http.CurlResponseHeaderReader.ReadStatusLine(HttpResponseMessage response)
   at System.Net.Http.CurlHandler.MultiAgent.CurlReceiveHeadersCallback(IntPtr buffer, UInt64 size, UInt64 nitems, IntPtr context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Net.Http.HttpClient.<FinishSendAsync>d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ConsoleApp2.Program.Main(String[] args)
```

My `Microsoft.NETCore.App` version is `1.1.0`
And my code like this:
```c#
using System;
using System.Net.Http;

namespace ConsoleApp2
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var client = new HttpClient();
            var request = new HttpRequestMessage(HttpMethod.Get, "https://http2.akamai.com/");
            request.Version = Version.Parse("2.0");
            
            var response = client.SendAsync(request).ConfigureAwait(false).GetAwaiter().GetResult();
            Console.WriteLine(response.Version.ToString());
            Console.ReadLine();
        }
    }
}
```

What's the problem?

**Comment by stephentoub** [comment]
I believe this is the same issue as https://github.com/dotnet/corefx/issues/14715, which was fixed for 2.0 in https://github.com/dotnet/corefx/pull/15203.

**Comment by karelz** [comment]
@Priya91 can you please confirm and close it as dupe?

**Comment by Priya91** [comment]
Yes, comparing the stack trace, the issue reported by @yaakov-h in the later part of the issue thread is the same as this one. @Jamesxql this is fixed for 2.0, you can get the latest fixed lib from dotnet-core myget feed [here](https://dotnet.myget.org/gallery/dotnet-core). Also note with 2.0, we don't have individual packages or project.json. So you'll be referencing a single Microsoft.NetCore.App package.
